File: ctlweb/src/graph/make-browser-invocation.ts
import { gql, useMutation } from 'urql'
import { Query } from './types'

const query = gql`
  mutation ($invocation: BrowserInvocationInput!) {
    makeBrowserInvocation(invocation: $invocation)
  }
`

export const useMakeBrowserInvocation = () => useMutation<Query>(query)[0]

File: internal/serve/mutation/make_browser_invocation.go
package mutation

import (
	"context"

	"github.com/enuesaa/walkhttp/internal/invoke"
	"github.com/enuesaa/walkhttp/internal/serve/schema"
)

func (r *MutationResolver) MakeBrowserInvocation(ctx context.Context, invocation schema.BrowserInvocationInput) (bool, error) {
	invokeSrv := invoke.New(r.Repos)
	entry := invocation.ToEntry()

	if err := invokeSrv.Write(entry); err != nil {
		return false, err
	}
	return true, nil
}
